packages/jest-circus/src/eventHandler.ts

```diff
@@ -182,11 +177,11 @@ const eventHandler: EventHandler = (event, state): void => {
       break;
     }
     case 'teardown': {
-      invariant(state.originalGlobalErrorHandlers);
-      invariant(state.parentProcess);
+      invariant(state.originalGlobalErrorHandlers, '');
+      invariant(state.parentProcess, '');
       restoreGlobalErrorHandlers(
-        state.parentProcess,
-        state.originalGlobalErrorHandlers,
+        state.parentProcess!,
```

Comment:
should they be `| undefined` in `restoreGlobalErrorHandlers` instead?

Reply:
`restoreGlobalErrorHandlers` does not checks for `undefined` values

https://github.com/facebook/jest/blob/d2023f120856314d5c3ca5b2661f894fda471384/packages/jest-circus/src/globalErrorHandlers.js#L28-L35

Reply:
Their types are also correct as they are not assigned to `INITIAL_STATE`

Reply:
the types must be wrong at some level since you add `!` and we don't get runtime errors from it

Reply:
ok, I checked this out. The reason they are possibly `undefined` is that they are first assigned during `setup`. By the time we come to `teardown` they are set, but we cannot statically verify that